api/base/views.py

```diff
@@ -437,7 +460,27 @@ def root(request, format=None, **kwargs):
     if utils.has_admin_scope(request):
         return_val['meta']['admin'] = True
 
-    return Response(return_val)
+    resp = Response(return_val)
+
+    # This is used exclusively for our partnership with Sloan and can be deleted after their study is complete.
+    referer_url = request.environ.get('HTTP_REFERER', '')
+    if DOMAIN + 'preprints/' in referer_url:
+        provider = PreprintProvider.objects.get(_id=referer_url.split('/')[4])
```

Comment:
Is this going to work for branded domains?

Reply:
No :(